TradeEngine/src/main/java/com/gala/sam/tradeEngine/utils/MarketUtils.java

```diff
@@ -20,27 +25,30 @@
 @Slf4j
 public class MarketUtils {
 
-  public static <T extends ActiveOrder> void queueIfTimeInForce(T order,
-      SortedSet<T> sameTypeLimitOrders, Consumer<Order> saveOrder) {
-    if (order.getTimeInForce().equals(TIME_IN_FORCE.GTC)) {
+  public static <T extends AbstractActiveOrder> void queueIfTimeInForce(T order,
+      SortedSet<T> sameTypeLimitOrders, Consumer<AbstractOrder> saveOrder)
+      throws OrderTimeInForceNotSupportedException {
+    if (order.getTimeInForce().equals(TimeInForce.GTC)) {
       log.debug("Time in force is GTC so add to queue");
       sameTypeLimitOrders.add(order);
       saveOrder.accept(order);
-    } else if (order.getTimeInForce().equals(TIME_IN_FORCE.FOK)) {
+    } else if (order.getTimeInForce().equals(TimeInForce.FOK)) {
       log.debug("Time in force is FOK so drop");
     } else {
-      throw new UnsupportedOperationException("TIME IN FORCE mode not supported");
+      throw new OrderTimeInForceNotSupportedException(order.getTimeInForce());
```

Comment:
Replace exception with logging.

Reply:
Please replace this exception with logging.

The exception is not needed, because the logic of the method is not violated. Judging from the name of the method, its purpose is "queue order if time in force is GTC". If time in force is not GTC, then it should simply not queue the order. If the time in force is weird or unrecognised, you should log a warning (because this means you need to talk to your upstream data source). However, it is _no reason_ to throw an exception and stop the method from returning.

The general rule is _only throw an exception if the method can not return a logically coherent result_. (And then catch and handle that exception as early as possible.) Please review all of the places that you throw exceptions, and remove the exceptions that are unnecessary.